src/xbot/common/controls/actuators/wpi_adapters/CANTalonWPIAdapter.java

```diff
@@ -477,4 +477,11 @@ public void updateTelemetryProperties() {
         temperatureProperty.set(this.getTemperature());
     }
 
+    @Override
+    public void ensureModeForTalon(TalonControlMode mode) {
+        if (this.getControlMode() != mode) {
+            this.setControlMode(mode);
```

Comment:
The fact that the two implementations are so different seems odd. What if we made `XCANTalon` into an abstract class?